KWB Easyfire support

bimbar opened the pull request:
**Description:**

KWB Easyfire pellet central heating units with the Comfort3 controller (http://www.kwbheizung.de/de/produkte/kwb-comfort-3.html) have a RS485 port over which sensor and state information is relayed.
This component supports direct connection via serial or via telnet terminal server. The serial cable has to be attached to the control unit port 25 (which is normally used for detached control terminals).
Since this serial protocol is proprietary and closed, only most temperature sensors and a few control relais are supported, the rest is still WIP (see https://www.mikrocontroller.net/topic/274137).

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

**Example entry for `configuration.yaml` (if applicable):**
```yaml
  - platform: kwb
    host: <ip>
    port: 23
    type: tcp
    raw: False
```
or
```yaml
  - platform: kwb
    device: "/dev/ttyUSB0"
    type: serial
    raw: False
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

balloob committed:
Merge pull request #5530 from home-assistant/dev

0.37

DavidLP committed:
Fixes: Pilight Switch rejects alphanumeric IDs #5119 (#5601)

Danielhiversen committed:
bug fix in hue (#5623)

tchellomello committed:
Fixes issue #5627 by bumping external Amcrest module to version 1.1.4 (#5662)

Add an optional extended description…

molobrakos committed:
Upgraded tellduslive (#5664)

pvizeli committed:
Bugfix sonos group coordinator (#5691)

* Bugfix sonos group coordinator

* Fix tests

pvizeli committed:
Bugfix async blocking loop with xml parser. (#5694)

jabesq committed:
Hotfix for netatmo cameras (#5644)

* Fix for missing netatmo tags in 0.37

Also fix issue with SSL certificate for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Netatmo welcome: vpn_url can be empty

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* add config floag to disable SSL verification for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Import CONF_VERIFY_SSL from const

pvizeli committed:
Fix bug for UNREACH devices / Variable handling and update. (#5689)

* Fix bug for UNREACH devices / Variable handling and update.

* fix track_time

* update after data after creation

* add message output

* change unreach

* change unreach code

* Revert "change unreach code"

This reverts commit f58430de3cb854d5f6f886bfe4d59899c1efadbf.

* update pyhomematic

pvizeli committed:
Bugfix sonso source input (#5699)

turbokongen committed:
[lock.zwave] Bugfix Zwave lock  (#5619)

* Bugfix state

* remove debug

aequitas committed:
Prevent infinite loop in crossconfigured mqtt event streams (#5624)

* Prevent events about MQTT messages received to cause infinite loop when two HA instances are crossconfigured for mqtt_eventstream.

* Fix linting

* Publish all MQTT received events except incoming from eventstream. Also make it configurable.

tboyce021 committed:
Fix hue lightgroups not syncing state (#5702)

balloob committed:
Version bump to 0.37.1

balloob committed:
Merge pull request #5707 from home-assistant/release-0-37-1

0.37.1

benvm committed:
Modbus write_register accept list

bimbar committed:
KWB Easyfire support

bimbar committed:
lint fixes

bimbar committed:
lint fixes

pvizeli committed:
Add unittest for multible entities in one line (#5759)

MartinHjelmare committed:
Update mysensors sensors (#5764)

* Add V_DIRECTION type for S_WIND.
* Add more predefined units of measurement.

Yannic-HAW committed:
- added send packet service to send ir packets directly from scripts (no need to define a switch for each command) (#5768)

- changed ip suffix of SERVICE_LEARN to use _ instead of . as ip seperator to avoid errors if used in yaml

Danielhiversen committed:
flux led o.13 (#5774)

aequitas committed:
Make brightness display work for rgb devices. (#5675)

* Make brightness display work for rgb devices.

Non rgbw devices return 255 for getWarmWhite255. This is part 2 to make the brightness slider work for these devices.

https://github.com/Danielhiversen/flux_led/pull/25

* Query brightness property which return WW level or RGB brightness equivalent.

https://github.com/Danielhiversen/flux_led/pull/25

andrey-git committed:
Merge pull request #5718 from benvm/modbus_write_register_array

Modbus write_register doesn't accept list

adrianlzt committed:
Allow to use data for enhanced messages (#5763)

Add notification data field to the message send to Facebook.
Allows to construct richer messages like cards, quick replies, attach
images, videos, etc

albertoarias committed:
Add support for position on wink cover (#5751)

Add support for position property for wink covers and fix state when stopped in the middle.

armills commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+CONF_RAW = 'raw'
+
+SERIAL_SCHEMA = {
+    vol.Required(CONF_PORT): cv.string,
```
Instead of `CONF_PORT` for serial mode, can you use `CONF_DEVICE` from homeassistant.conf? That should be a little less confusing that reusing the same config field, and then you won't need to specify the type, since you can check if either `host` or `device` is specified.

bimbar replied:
Sure, but I want to keep the explicit switch tcp/serial.

armills commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+
+ETHERNET_SCHEMA = {
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.positive_int,
```
Small change, for `CONF_PORT`, you can use `cv.port`.

bimbar replied:
ok

armills commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+    vol.Optional(CONF_TYPE): 'raw',
+}
+
+
```
You need to define `PLATFORM_SCHEMA` here. Right now the schemas you have defined aren't being used.

[Use the example here](https://home-assistant.io/developers/platform_example_light/) as a guide on how to get started. Since you can connect via TCP or serial, the best way is to have `CONF_HOST` and `CONF_DEVICE` be part of an exclusive group. [(Example here)](https://github.com/home-assistant/home-assistant/blob/9cad9c19f8a29c0173cb7672fd3ad95d91c34ee1/homeassistant/components/keyboard_remote.py#L39). This will make sure that only one of the two is defined, and will make it so the user doesn't need to add a redundant type config.

bimbar replied:
I'm highly dubious if what I've done here works, but it should be considerably closer.

armills commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+
+    from pykwb import kwb
+
+    _LOGGER.info('KWB: initializing')
```
The logger will add a kwb prefix already based on the filename, so the "KWB: " on all of these logs can be dropped. Just helps keep it a little cleaner. :)

bimbar replied:
ok

armills commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+    _LOGGER.info('KWB: starting thread')
+    easyfire.run_thread()
+    _LOGGER.info('KWB: thread started')
+
```
Add a handler here to stop the thread as well when home assistant exits. [There's an example here.](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/binary_sensor/flic.py#L69) In this case, instead of `client.close()` you'll have `easyfire.stop_thread()`.

bimbar replied:
I close the thread in pykwb on destruct, which should be ok, too, but this is probably the cleaner way to do it.

armills replied:
It certainly doesn't hurt, but this makes sure the thread is stopped cleanly while hass is shutting down.

bimbar replied:
ok

emlove reviewed (changes requested):
Nice work! Early support is fine, since it can always be extended later.

One note in addition to the comments: If you expose a callback from pykwb when a sensor gets updated, you can immediately update the state in home assistant by calling `schedule_update_ha_state()` on the entity. [Example](https://github.com/home-assistant/home-assistant/blob/89ec752064709358e2e49c6a51c39841a09da5e8/homeassistant/components/tellduslive.py#L165). This isn't required, but could be an easy addition.

bimbar commented:
I'll have to go over the whole validation stuff, I copy&pasted it and hoped for the best ;) .

I don't think the on demand updates are applicable here, the heating unit blasts these out over the serial port as fast as it can, so I could schedule about 10 updates a second with the data I have. Polling the current state at regular intervals seems the better way to do it.

pvizeli committed:
Migrate lock component to async (#5748)

armills commented:
Yeah, it's no problem. Having to pick between tcp/serial makes the validation definition extra tricky. Let me know if you have any questions.

Makes sense that you don't want to flood hass with updates. As an FYI, right now it'll poll with the [sensor default of 30 seconds](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/__init__.py#L15). You can override it for this platform by copying that line into your platform code and specifying an alternate default update interval. [These both can also be overwritten by the user in the platform config by setting `scan_interval`.](https://home-assistant.io/topics/platform_options/#scan-interval) Given that it's a heating unit though, it probably doesn't need to be updated very frequently.

andriej committed:
Update package.py (#5783)

tchellomello committed:
Fix attribute error for media_player/roku if roku device is unreachable and shows a persistent notification (#5785) (#5786)

ray0711 committed:
Fix brightness slider for mqtt template lights (#5780)

Add the missing supported_features declaration.

sbidoul committed:
add min and max jinja filters (#5765)

pvizeli committed:
Sonos fix favorite, coordinator, cleanup update (#5778)

* Sonos fix favorite, coordinator, cleanup update

* Bugfix snapshot restore

alexmogavero committed:
Improve Sky Hub error handling (#5762)

* Added error handling in function _get_skyhub_data

* Error line split for readability

Commit:
Added error checking to the MIMEImage encoding for smtp.py (#5753)

* Added error checking to the MIMEImage encoding for smtp.py

Added fallback to file attachment rather than inline image for images
without a known MIME

* PEP8 reqs to fix previous commit

postlund committed:
Initial support for Apple TV (#5698)

* Initial support for Apple TV

* hash_wip

* Add media_play support to Apple TV

herm committed:
Check for command topics when determinig the capabilities of an MQTT light. (#5770)

Previous code used the state topic which is obviously wrong:
- The state topic is already used to select optimistic mode.
- A light with only the state topic but no command topic would still announce the capability.

andrey-git committed:
Check config before restarting (#5609)

* Check config before restarting.

* Make check_config on restart async

* don't check if notification service exists

* Use .communicate()

* Reduce the number of notifications. Add tests.

tboyce021 committed:
Fix Hue groups with same names (#5737)

* Revert "Fix hue lightgroups not syncing state (#5702)"

* Use light_id in unique_id for Hue groups

* Make sure HueLight unique_id is unique

* Update hue.py

* Update hue.py

* Update hue.py

tboyce021 committed:
Add icon_template to template sensor (#5766)

* Add icon_template to template sensor

* Update test_template.py

* Update test_template.py again

* Update template.py

* Update test_template.py

* Update test_template.py

robbiet480 committed:
TTS ID3 support (#5773)

* Add support for writing ID3 tags to the file for improved display in media players

* Lint and async fixes

* Use mutagen instead of taglib

* Fix tests

* Add fallback for album

* Requested changes

* move import

* Fix album name

* Change default options handling

* Move to member function / minor fix

* fix style

* fix lint

* change mutagen handling

* fix lint / add name to bytesio

* Update __init__.py

* Fix test, some cleanups

* Add mutagen exeption handling, fix tests

* fix mutagen taging

fabaff committed:
MQTT discovery (#5724)

* Change implementation

* Re-write

* Remove unused consts

* Update discovery.py

* Add tests

* fix other tests

* Fix check_config script test

* Lint

* Lint

MartinHjelmare committed:
Add mysensors device tracker and platform discovery (#5781)

* Add mysensors device_tracker platform

* Add discovery of device_tracker platforms

* Enable discovery of device_tracker platforms that are not
  DeviceScanner.
* Update signature of setup_scanner function in all affected platforms.
* Add test.
* Use discovery for mysensors device_tracker platform.

* Remove gps accuracy

* Small change to core like schema

* fix depency

pvizeli committed:
Revert last change on TTS cache load for more speed (#5797)

fabaff committed:
Upgrade pyasn1 to 0.2.2 (#5796)

JshWright committed:
Update python-nest dependency version (#5795)

The nest-cam changes have now been merged into the upstream library, so
there is no need to track a specific branch.

Updating to 3.1.0 also fixes a structure parsing bug I was experiencing.

robbiet480 committed:
Bump netdisco to 0.8.2

andrey-git committed:
Create a file for zwave workarounds. (#5798)

* Create a file for zwave workarounds. Add sensor->binary_sensor for fgfs101 (#2)

* Don't use default None

jawilson committed:
Fix climate.set_fan_mode yaml (#5799)

emlove committed:
Formalize supported_features as entity property (#5794)

* Formalize supported_features as entity property

* Remove extra emulated_hue conditions

* Generate log message in executor

balloob committed:
Update frontend (#5800)

kellerza committed:
WIP: [component/recorder] Refactoring & better handling of SQLAlchemy Sessions (#5607)

* Refactor recorder and Sessions

* Cover #4352

* NO_reset_on_return

* contextmanager

* coverage

aequitas committed:
Add support for fluxled discovery. (#5784)

* Add support for fluxled discovery.

* Make use of device type/protocol auto detection.

pschmitt committed:
Mailgun notify service (#5782)

* Mailgun notify service

* Update dependency to version 1.3

- The provided credentials (including the domain) are now checked during
startup, as requested by @balloob
- The domain name is now optional
- There's a new config item "sandbox" which indicates whether to use the
sandboxed domain in case the domain is not set

* Fix a few lint issues

* Disable lint check no-value-for-parameter

postlund committed:
Add discovery suppport to Apple TV (#5801)

Add an optional extended description…

nodinosaur committed:
The word `router` was misspelt (#5803)

fixed as this message is user facing

kellerza committed:
Update sma.py (#5807)

Danielhiversen committed:
Improve warning message in template rendering (#5806)

* improve warning message when template is none

* improve error message when template is none

* improve error message when template is none

* improve error message when template is none

pvizeli committed:
Move core service from core to components (#5787)

* Move core servcie from core to components

* add new handler for signals/exception

* Static persistent id

* Move unittest

* fix coro/callback

* Add more unittest for new services

* Address comments

* Update __init__.py

pvizeli committed:
[Image_Processing][Breaking Change] Cleanup Base face class add support for microsoft face detect (#5802)

* [Image_Processing] Cleanup Base face class add support for microsoft face detect

* fix lint

* add unittest for micosoft detect

* fix test

andrey-git committed:
Make sure workaround_component is not none (#5808)

rytilahti committed:
Add available property and typing hints (#5593)

* light.demo: add available property, add typing hints

* light.demo: keep all lights available, fix init ordering

* Fix issues raised during review

* Update demo.py

acambitsis committed:
Upgrade Russound integration to v0.1.7 (#5756)

* Bumped up version to use 0.1.7 of Russound integration module.
Fixed bug arising from not supporting TURN_ON state (fixes issue https://github.com/home-assistant/home-assistant/issues/5012)
Implemented state support in 0.1.7 such that component state is returned from the actual AMP. (Still uses polling model though).
Tested it with home-assitant users @laf (original developer of the module) and @hofsta.  Works fine with their Russounds.

* Made styling / compliance changes and updated correct version of russound module on requirements_all.txt.

* Changed handling of properties to be compliant with https://github.com/home-assistant/home-assistant/issues/4210
(Specifcailly added member variables for state, volume and source to cache these values, and introduced Update() method to set their values).

Now returns None if the selected source index that is returned from russound is greater than the length of the specified source list in the yaml config.
Removed unnecesary comment.

* Removed blank line after docstring.

* Removed updated() in class init and added True paramter to add_devices in setup_platform.

* Dropped the no longer needed self.update()

kellerza committed:
Initial (#5811)

pvizeli committed:
Ffmpeg update 1.4 (#5813)

* Pump ffmpeg version

* update entity

* next 1.4

Teagan42 committed:
Change medium state for filtering (#5817)

sdague committed:
Enhancements to ARWN platform (#5816)

* Fix arwn platform discover_sensors

The discover_sensors function can return either singletons or a list
of sensors. However the consumer was always expecting a list. This
fixes it to work in both cases.

* Add custom icons to arwn sensors.

This adds some custom icons for different kinds of weather sensors
that the arwn platform returns. Makes it a little easier to see what's
going on.

Commit:
Fixed call to object_id() (#5814)

kellerza committed:
[recorder] Protect against running in the event loop (#5812)

turbokongen committed:
Bugfixes (#5740)

* wrong data for lock alarm_type

* missing whitespace

* Not possible to set codes starting with 0

Duoxilian committed:
Support away_mode as permanent hold and hold_mode as temporary hold. (#5725)

* Support away_mode as permanent hold and hold_mode as temporary hold.

* Add comments to explain code better. Remove indefinite hold preference
to be consistent with 'away_mode'.

balloob committed:
Remove non-working webfont preload (#5819)

balloob committed:
Update frontend (#5820)

balloob committed:
Upgrade aiohttp to 1.3 (#5821)

pvizeli committed:
Move signal handling out of core to bootstrap (#5815)

* Move signal handling out of core to bootstrap

* Fix tests

balloob committed:
Deprecate event decorators (#5822)

balloob committed:
Make device sun light trigger async (#5823)

turbokongen committed:
Adding helper for get and set values (#5743)

* cleanup

* Update __init__.py

* Update __init__.py

* Update __init__.py

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

molobrakos committed:
Don't thow exception if connection to server is lost (#5775)

molobrakos committed:
Don't thow exception if connection to server is lost (#5775)

balloob committed:
Convert config.components to a set (#5824)

postlund committed:
Handle connection errors when connecting to Apple TVs (#5829)

* Handle connection errors when connecting to Apple TVs

Also bump pyatv to 0.1.2 which fixes a request leak.

* Fix pylint error

* Fix import order

postlund committed:
Handle connection errors when connecting to Apple TVs (#5829)

* Handle connection errors when connecting to Apple TVs

Also bump pyatv to 0.1.2 which fixes a request leak.

* Fix pylint error

* Fix import order

scipioni committed:
Telegram webhooks (#5793)

* new component telegram_webhooks

* keyboard support in telegram notify

* telegram_webhooks has no tests

* requirement like notify/telegram

* ops, requirements_all.txt needed for travis

* ops, requirements_all.txt is generated by script/gen_requirements_all.py

* check telegram trusted networks in web handler

* raise an event now

* use of hass.config.api.base_url

* more readable

* small cleanups

* Small style change for HA guideline

* fix lint

* revert return to origin

postlund committed:
Reuse default aiohttp session (#5836)

postlund committed:
Reuse default aiohttp session (#5836)

pvizeli committed:
Update aiohttp 1.3.1 (#5838)

pvizeli committed:
Update aiohttp 1.3.1 (#5838)

pschmitt committed:
Refactoring and JSON decode error handling (#5826)

* Refactoring and JSON decode error handling

* Catch ValueError instead of simplejson.scanner.JSONDecodeError

pschmitt committed:
Refactoring and JSON decode error handling (#5826)

* Refactoring and JSON decode error handling

* Catch ValueError instead of simplejson.scanner.JSONDecodeError

emlove committed:
[recorder] Add tests for full schema migration (#5831)

* [recorder] Add tests for full schema migration

* Remove leftover code

* Fix duplicate creation of sqlalchemy Index object

* It's that kind of day...

* Improve models_original docstring

emlove committed:
[recorder] Add tests for full schema migration (#5831)

* [recorder] Add tests for full schema migration

* Remove leftover code

* Fix duplicate creation of sqlalchemy Index object

* It's that kind of day...

* Improve models_original docstring

bokub committed:
Add history_stats sensor (#5610)

* Add new sensor/history_stats component

* Add real unit tests

* Code style changes requested for pull request

* Remove time aliases & allow datetime values

* Reformat unit tests

* Remove all async behavior in history_stats

* Change duration format

markoudev committed:
Add support for multiple devices to Tado device tracker (#5810)

* Add support for multiple devices to Tado device tracker

* Hound did not agree with my usage of spaces

* Two blank lines were expected

* Keep CONF_HOME_ID in the Tado file

* Make home_id optional

* Work with different API URLs depending on home_id being present

* Remove CONF_HOME_ID from HA's const.py file

* Missed removing CONF_HOME_ID from the import

andrey-git committed:
Fix zwave helper getter not to fail on some None results. (#5845)

andrey-git committed:
Fix zwave helper getter not to fail on some None results. (#5845)

LinuxChristian committed:
D-Link switch version bump of external library (#5843)

LinuxChristian committed:
D-Link switch version bump of external library (#5843)

rytilahti committed:
bump python-yeelight version (#5850)

Add an optional extended description…

rytilahti committed:
bump python-yeelight version (#5850)

Add an optional extended description…

konikvranik committed:
version bump (#5846)

Add an optional extended description…

pvizeli committed:
telegram small cleanup/style (#5852)

pvizeli committed:
Fix check_config script. (#5853)

pvizeli committed:
Fix check_config script. (#5853)

andrey-git committed:
Add support for zwave wakeup setting. (#5849)

* Add support for zwave wakeup setting.

* rename wakeup

molobrakos committed:
Do not call state if device isn't available (#5835)

balloob committed:
Core: cleanup timer (#5825)

* Minor core cleanup

* Cleanup timer

* Lint

* timeout with correct loop

* Improve timer thanks to pvizeli

* Update core.py

* More tests

balloob committed:
Update frontend (#5855)

balloob committed:
Update frontend (#5855)

kellerza committed:
[recorder] Run end model changed in session scope (#5858)

kellerza committed:
[recorder] Run end model changed in session scope (#5858)

balloob committed:
Recorder run can be None (#5854)

balloob committed:
Recorder run can be None (#5854)

aequitas committed:
Rflink update and small refactor. (#5789)

* Use same pattern for device defaults in both platforms.

* Update Rflink that passes loop downstream.

* Update requirements.

aequitas committed:
Rflink update and small refactor. (#5789)

* Use same pattern for device defaults in both platforms.

* Update Rflink that passes loop downstream.

* Update requirements.

emlove committed:
binary_sensor sensor_class to entity device_class (#5860)

* binary_sensor sensor_class to entity device_class

* Linter fixes

* Should be it

tchellomello committed:
Avoid traceback for Amcrest cameras/firmware that does not have the software_information API call (#5865)

* Avoid traceback for Amcrest cameras/firmware that does not have the software_information API call

* Make lint happy

tchellomello committed:
Avoid traceback for Amcrest cameras/firmware that does not have the software_information API call (#5865)

* Avoid traceback for Amcrest cameras/firmware that does not have the software_information API call

* Make lint happy

balloob committed:
Version bump to 0.38

balloob committed:
Update frontend (#5866)

jjmontesl committed:
Adds play URL support to mpd (makes it work with tts). (#5863)

* Adds play URL support to mpd (makes it work with tts).

* Removed extra line at the end of the file.

Danielhiversen committed:
rfxtrx 0.17 (#5870)

konikvranik committed:
version bump (#5871)

fabaff committed:
Upgrade psutil to 5.1.3 (#5873)

andrey-git committed:
Stop using entity_picture that is known to be bad. (#5856)

* Stop using entity_picture that is known to be bad.

* Only abandon image on 400 or 404 response

* Return is_permanent_failure as a third part of response

* Add debug printout

* Fix lint

* Fix lint

fabaff committed:
Style updates (#5878)

* Use consts, update quoting, and update comments

* Update log message

* Update docstrings

* Update log messages

* Update log messages

* Update style

* Fix typo

* Add newline

* Fix indent

* Fix line too long

balloob committed:
Revert "Stop using entity_picture that is known to be bad." (#5882)

kellerza committed:
Component set add unsing OR (#5880)

robbiet480 committed:
Merge pull request #5833 from home-assistant/release-038

Release 0.38

robbiet480 committed:
Version bump to 0.39.0.dev0

balloob committed:
Add config component and hassbian example panel (#5868)

* Add hassbian panel

* Rename to generic config panel

* Allow loading hassbian as test

* Add tests

* Update frontend

* Lint

* Lint

balloob committed:
Upgrade AppleTV dep to 0.1.4

balloob committed:
0.38.1 (#5889)

* Upgrade AppleTV dep to 0.1.4

* Version bump to 0.38.1

* Update logbook in frontend

pschmitt committed:
[lock.nuki] Fix Nuki lock for Python 3.4 and 3.5 (#5899)

happyleavesaoc committed:
bump myusps version (#5911)

bachp committed:
[google] Update google component dependencies (#5904)

jjmontesl committed:
Fixed proximity zone incorrectly using name instead of zone setting. (#5862)

* Fixed proximity zone incorrectly using name instead of zone setting.

* Fixed proximity tests to suit change.

emlove committed:
Add device_class support to cover component (#5881)

pavoni committed:
Bump pywemo - fix for latest bridge firmware. (#5921)

fabaff committed:
Fix name (#5912)

andrey-git committed:
Make hdmi_cec work with the new customize (#5916)

balloob committed:
Add check_config API (#5898)

* Add check_config API

* Add config panel to default config

* Add tests

* Lint

* lint

balloob committed:
Update frontend (#5923)

postlund committed:
Add media_image to media_player component (#5754)

persandstrom committed:
vasttrafik: update token on read error (#5875)

colinodell committed:
Update qnapstats library; add support for new verify_ssl config (#5919)

happyleavesaoc committed:
ups sensor (#5890)

happyleavesaoc committed:
fedex sensor (#5891)

* fedex sensor

* fix fedex requirement

happyleavesaoc committed:
[WIP] gstreamer media player (#5839)

* gstreamer media player

* gstreamer meta

bimbar commented on homeassistant/components/sensor/kwb.py:
```diff
@@ -0,0 +1,101 @@
+SERIAL_SCHEMA = {
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(CONF_TYPE): 'serial',
+    vol.Optional(CONF_TYPE): 'raw',
```
ok

pvizeli committed:
Add 'wait_template' to script commands / Refactory track_template (#5827)

* Add 'wait' to script commands.

* Add track_template + unittest / rename wait_template

* fix lint & test

* Fix handling / change automation-template / add tests

* address paulus comments

bimbar committed:
Some CHanges

bimbar committed:
Some CHanges

postlund committed:
Fix artwork in Apple TV platform (#5874)

bimbar commented:
Latest commit probably doesn't lint anymore, but it's primarily for getting validation sorted out.

balloob committed:
Merge branch 'master' into dev

andrey-git committed:
force_update zwave sensors. (#5844)

* force_update zwave sensors with polling_intensity.

* use bool to cast

* Make all zwave sensors force_update=True

andrey-git committed:
Fix getters that ignored labels. (#5903)

* Fix getters that ignored labels.

* Try 5 times for changed dict

* fix lint

* Add decorator with retrying functions.

* Fix lint

* use retries instead of decorator

turbokongen committed:
Bugfix for #5900 (#5901)

Commit:
Fixed typos in wink climate (#5936)

pvizeli committed:
Core cleanup: two stage shutdown (#5876)

* Core cleanup: two stage shutdown

* fix spell

* fix

* add async logger to close

* change aiohttp to use CLOSE

* address paulus comments

* Fix tests

* Add unittest

balloob committed:
Z-Wave: Rename customize to device_config [Breaking change] (#5933)

* Z-Wave: Rename customize to device_config

* Fix light

* Make entity id not required

balloob committed:
Update frontend (#5939)

robbiet480 committed:
Fix AWS Lambda breakage after #5824 (#5935)

Add an optional extended description…

alanfischer committed:
iTach Remote Platform (#5660)

* Added itach remote device

* Added coverage & requirements

* Updated requirements

* Added schema, inline commands, and platform_setup failure if itach not found

* Removed unnecessary optional parameter

* Removed filename option for itach remote

arraylabs committed:
Added myq cover component (#5886)

Add an optional extended description…

fabaff committed:
Upgrade TwitterAPI to 2.4.4 (#5940)

MartinHjelmare committed:
Fix mysensors platforms version requirement (#5942)

* Notify and device tracker platforms require mysensors version 2.0 or
  greater.

fabaff committed:
Upgrade thingspeak to 0.4.1 and use the correct API key (#5906)

valentinalexeev committed:
Bugfix/waqi sensor pwaqi version bump (#5944)

* Bump pwaqi to 1.4 to fix a typo in the underlying library.

* Update WAQI sensor to use pwaqi 2.0 which relies on AQICN public API.
This is a breaking change as the component now requires 'token' parameter.

* Fix lint

pschmitt committed:
Reduce battery drain on Nuki Lock (#5945)

Cadair committed:
Add support for the Open Energy Monitor Thermostat (#5583)

* Add support for the Open Energy Monitor Thermostat

* Fix linting errors

* Define an update method and local state

* fix linter

* Small tweaks.

Update oemthermostat version, default name and docstrings

* Fail to setup oem platform if connection fails.

* update requirements

* More tweaks to auth and exceptions

* Remove target temp

balloob committed:
Fix Z-Wave (#5955)

andrey-git committed:
Add workaround for ignoring zwave devices and reversing cover open/close (#5922)

andersonshatch committed:
Only try to pair notify.webostv when not paired (#5967)

This allows the notify.webostv component to successfully register when
the TV is not currently powered on as long as a pairing record exists.

keerts committed:
Fix slow status updates from the knx bus (#5963)

* Fix slow status updates from the knx bus

The data set in the entity was an array, not the value.

@fixes https://github.com/home-assistant/home-assistant/issues/4407

* speed up status update

balloob committed:
Clean up HDMI_CEC [Breaking change] (#5932)

fabaff committed:
Remove unit of measurement

Commit:
Fix #5979 (#5980)

robbiet480 committed:
Merge pull request #5975 from fabaff/unit-moon

Remove unit of measurement

balloob committed:
Add initial Z-Wave config panel (#5937)

* Add Z-Wave config panel

* Add config to Z-Wave dependencies

* Lint

* lint

* Add tests

* Remove temp workaround

* Lint

* Fix tests

* Address comments

* Fix tests under Py34

andrey-git committed:
Allow printing the number of states returned by history and time it took to extract. (#5973)

balloob committed:
Z-Wave file name fix (#5985)

andrey-git committed:
Add 'days' flag to history fetch urls. (#5895)

* Add 'days' flag to history fetch urls.

* Fix unrenamed variable

* Switch to end_time param instead of days

* Checkthat end_time is parsed

balloob committed:
Fix Z-Wave node config (#5986)

balloob committed:
Update frontend (#5987)

fabaff committed:
Enable sensor for discovery (#5974)

* Enable sensor for discovery

* Remove blank line

PetePriority committed:
Added fritzbox_netmonitor.py (#5469)

* Added fritzbox_netmonitor.py

* Implemented changes

molobrakos committed:
install libsodium to enable decryption of encrypted owntracks payload (#5976)

happyleavesaoc committed:
timeMin fix (#5983)

Add an optional extended description…

soldag committed:
Added limitlessled support for bridge v6 and RGBWW bulbs. (#5958)

* Added limitlessled support for bridge v6 and RGBWW bulbs.

* Fix minor code style issue.

* Updated requirements_all.txt

fabaff committed:
Upgrade pylast to 1.8.0 (#5991)

bimbar committed:
hound-ci

bimbar committed:
hound-ci

bimbar committed:
rebase -> 0.38.2

bimbar committed:
Merge branch 'kwb' of https://github.com/bimbar/home-assistant into kwb

bimbar committed:
coveragerc

bimbar committed:
hound-ci

bimbar commented:
Maybe I shouldn't have done "catch up to reality". Please tell me if I should reopen this PR as a new one.

armills commented:
It's good to catch up to reality, but you need to rebase on dev instead of the release branch (0.38.2), since the release branch has diverging commits. You should be able to run `git rebase -i upstream/dev`. Adding `-i` will let you choose which commits to keep. Basically you can just delete every line this isn't one of your commits. Then you can push back to your pr branch using the force `-f` flag.

armills commented:
You can also just copy out your changes, make a new branch off of dev, and force push that branch to the PR. Whichever way is easier.

bimbar commented:
So I did bad things to git, please refer to
https://github.com/home-assistant/home-assistant/pull/6018
in the future.
